app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -87,67 +81,157 @@ export default class List extends PureComponent {
                 renderItem: this.renderUnreadItem,
-                topSeparator: false,
-                bottomSeparator: true,
-            });
-        }
-
-        if (favoriteChannelIds.length) {
-            sections.push({
+            };
+        case SidebarSectionTypes.FAVORITE:
+            return {
                 id: 'sidebar.favorite',
                 defaultMessage: 'FAVORITES',
-                data: favoriteChannelIds,
-                topSeparator: unreadChannelIds.length > 0,
-                bottomSeparator: true,
-            });
+            };
+        case SidebarSectionTypes.PUBLIC:
+            return {
+                action: this.goToMoreChannels,
+                id: 'sidebar.channels',
+                defaultMessage: 'PUBLIC CHANNELS',
+            };
+        case SidebarSectionTypes.PRIVATE:
+            return {
+                action: canCreatePrivateChannels ? this.goToCreatePrivateChannel : null,
+                id: 'sidebar.pg',
+                defaultMessage: 'PRIVATE CHANNELS',
+            };
+        case SidebarSectionTypes.DIRECT:
+            return {
+                action: this.goToDirectMessages,
+                id: 'sidebar.direct',
+                defaultMessage: 'DIRECT MESSAGES',
+            };
+        case SidebarSectionTypes.RECENT_ACTIVITY:
+            return {
+                action: this.goToDirectMessages,
+                id: 'sidebar.types.recent',
+                defaultMessage: 'RECENT ACTIVITY',
+            };
+        case SidebarSectionTypes.ALPHA:
+            return {
+                action: this.showCreateChannelOptions,
+                id: 'mobile.channel_list.channels',
+                defaultMessage: 'CHANNELS',
+            };
+        default:
+            return {
+                action: this.showCreateChannelOptions,
+                id: 'mobile.channel_list.channels',
+                defaultMessage: 'CHANNELS',
+            };
         }
+    }
 
-        sections.push({
-            action: this.goToMoreChannels,
-            id: 'sidebar.channels',
-            defaultMessage: 'PUBLIC CHANNELS',
-            data: publicChannelIds,
-            topSeparator: favoriteChannelIds.length > 0 || unreadChannelIds.length > 0,
-            bottomSeparator: publicChannelIds.length > 0,
-        });
+    buildSections = (props) => {
+        const {
+            orderedChannelIds,
+        } = props;
 
-        sections.push({
-            action: canCreatePrivateChannels ? this.goToCreatePrivateChannel : null,
-            id: 'sidebar.pg',
-            defaultMessage: 'PRIVATE CHANNELS',
-            data: privateChannelIds,
-            topSeparator: true,
-            bottomSeparator: privateChannelIds.length > 0,
+        return orderedChannelIds.map((s, i) => {
+            return {
+                ...this.getSectionConfigByType(s.type),
+                data: s.items,
+                topSeparator: i !== 0 && orderedChannelIds[i - 1].items.length > 0,
```

Comment:
Beforem `bottomSeparator` was dependent on effectively `s.items.length > 0`. I'm assuming that `orderedChannelIds` excludes empty sets? If so, why do we have to check the previous set's length to compute `topSeparator`? Can we just set it to `i !== 0`?